### PR TITLE
feat: Screenshots now have option to include UI elements.

### PR DESCRIPTION
**Pull request by astrid-kg**
#### Type of change

Feat  

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/REV-555
https://cognitedata.atlassian.net/browse/REV-581

## Description :pencil:

- `getScreenshot` now has the option to include UI elements in the returned image and will do so by default.
- `includeUI` parameter has been added to `getScreenshot` to allow screenshots that only contain rendered 3D.
- A "Screenshot" folder has been added to the GUI menu in the examples project.

## How has this been tested? :mag:

This was tested using the new menu option in the examples project. Tested with and without UI elements, at different resolutions and on screens with different pixel ratios.

## Test instructions :information_source:

Try out the new "Screenshot" option in the example project.

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [ ] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [x] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [x] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [x] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [x] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [x] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: viewer/packages/api/src/public/migration/Cognite3DViewer.ts [hunk 1]
```diff
@@ -1059,31 +1061,65 @@ export class Cognite3DViewer {
    * const url = await viewer.getScreenshot();
    * const image = document.createElement('img');
    * image.src = url;
-   * document.body.appendChild(url);
+   * document.body.appendChild(image);
```
Comment (larsmoa) [inline viewer/packages/api/src/public/migration/Cognite3DViewer.ts:1064]: Should there be some information about how HTML elements are handled and any corner cases? Maybe also extend documentation site with information and example

Comment (astrid-kg) [reply viewer/packages/api/src/public/migration/Cognite3DViewer.ts:1067]: I added some documentation and one example. 💪

File: documentation/docs/examples/cad-basic.mdx [hunk 2]
```diff
@@ -83,6 +83,27 @@ viewer.setBackgroundColor({ alpha: 0.0 });
 viewer.domElement.style.backgroundImage = "url('https://i.imgur.com/8tcxHWh.jpeg')";
 ```
 
+## Creating screenshots
+
+Use `getScreenshot` to create a screenshot from the current camera position. By default the resolution of the image is set to match the size of the render canvas, but a custom resolution can be provided.
+Whether or not to include UI in the screenshot can also be specified. When drawing UI, only the viewer DOM element and its children will be included in the image. The DOM is scaled to fit any provided resolution, 
+and as a result some elements can be positioned incorrectly in regards to the 3D render. 
+
+Note that `html2canvas` is used to draw UI and this has some limitations on what CSS properties it is able to render. For details see https://html2canvas.hertzen.com/documentation.
```
Comment (larsmoa) [inline documentation/docs/examples/cad-basic.mdx:92]:
```suggestion
Note that `html2canvas` is used to draw UI and this has some limitations on what CSS properties it is able to render. For details see [the html2canvas documentation](https://html2canvas.hertzen.com/documentation).
```

File: viewer/packages/api/src/public/migration/Cognite3DViewer.ts [hunk 3]
```diff
@@ -1053,9 +1054,13 @@ export class Cognite3DViewer {
   }
 
   /**
-   * Take screenshot from the current camera position.
+   * Take a screenshot from the current camera position. When drawing UI, only the viewer DOM element and its children will be included in the image.
+   * The DOM is scaled to fit any provided resolution, and as a result some elements can be positioned incorrectly in regards to the 3D render.
+   *
+   * `html2canvas` is used to draw UI and this has some limitations on what CSS properties it is able to render. See their {@link https://html2canvas.hertzen.com/documentation} for details.
```
Comment (larsmoa) [inline viewer/packages/api/src/public/migration/Cognite3DViewer.ts:1060]:
```suggestion
   * `html2canvas` is used to draw UI and this has some limitations on what CSS properties it is able to render. See {@link https://html2canvas.hertzen.com/documentation the html2canvas documentation} for details.
```

**Review by larsmoa — APPROVED** [review]
Looks great! 

Would consider to add a example for this in docs (separate PR).

**Pull request merged**